### PR TITLE
Add support for nested routers

### LINKED ISSUE #20: Bug with dynamic route and nested routers
For some reason, I'm getting strange error while using this package with a nested router on a dynamic route. This is the error:

```
TypeError: split(...).join is not a function
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:47
    at Array.forEach (<anonymous>)
    at iterateStack (/path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:7
    at Array.forEach (<anonymous>)
    at iterateStack (/path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:7
    at Array.forEach (<anonymous>)
    at iterateStack (/path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /path/to/node_modules/@wesleytodd/openapi/lib/generate-doc.js:17:5
```

The base router defines a dynamic path and requires a nested router to handle that path. It looks more or less like this:

```js
// router.js
const routes = require('express').Router();

routes.use('/:id', require('./_id'));

module.exports = routes;
```

The OpenAPI spec is used as a middleware on that nested router's base path like so:

```js
// _id.js
const router = require('express').Router({ mergeParams: true });
const { openapi } = require('../../../middleware');

router.get(
  '/',
  openapi.validPath({
    summary: 'Get a user.',
    parameters: [
      {
        in: 'path',
        imageId: 'id',
        schema: {
          type: 'integer',
        },
      },
    ],
    responses: {
      200: {
        content: {
          'application/json': {
            schema: {
              type: 'string',
            },
          },
        },
      },
    },
  }),
  async (req, res) => {
    res.send('done')
  }
);

module.exports = router;
```

If I completely remove the OpenAPI references, the application works as expected. The application also works as expected if I define the dynamic route handler in the same file as the router. It seems to only have an issue if I try using a nested router.

### PR DESCRIPTION
**Pull request by Megapixel99**
Fixes #20 and adds a test.

### CODE REVIEW

File: lib/generate-doc.js [hunk 1]
```diff
@@ -82,8 +82,25 @@ function iterateStack (path, routeLayer, layer, cb) {
   layer.route.stack.forEach((l) => iterateStack(path + layer.route.path, layer, l, cb))
 }
 
+function processComplexMatch (thing, keys) {
```
Comment (wesleytodd) [inline lib/generate-doc.js:85]: I think we need some comments here. This is looking for the key and then replacing it with the key template syntax?  There are a bunch of replaces there I am not sure I am following. An approach I like to use is to pull all the regex's out and name them variable that explain what they match. Ideally outside of the function scope so they are recreated each call for no reason.

**Review by wesleytodd — CHANGES_REQUESTED** [review]
I *think* this is good, but I would really like to make sure I understand all the regex operations before I approve.

File: lib/generate-doc.js (hunk 1 above)
Comment (wesleytodd) [reply lib/generate-doc.js:85]: Yeah, that was my thinking (add it to code comments). I think I followed most of this but it takes a bit and comments and/or var names would go along way to making this maintainable longer term and for folks who might not be as deep into it as we are.

Comment (wesleytodd) [reply lib/generate-doc.js:85]: Mind adding one over the middle block just to cover all of them?

**Comment by Megapixel99** [comment]
I removed a few lines of code that were not doing anything and reformatted the code, it should be a little cleaner

**Comment by Megapixel99** [comment]
Streamlined the process with less code

**Review by enov — APPROVED** [review]
LGTM!

**Review by wesleytodd — APPROVED** [review]

**Pull request merged**